specification/trace/semantic_conventions/instrumentation/elasticsearch.md

```diff
@@ -0,0 +1,41 @@
+# Semantic conventions for Elasticsearch
+
+**Status**: [Experimental](../../../document-status.md)
+
+This document defines semantic conventions to apply when instrumenting requests to Elasticsearch. They map Elasticsearch
+requests to attributes on a Span. Note that there may also be an http span created in addition to the Elasticsearch
```

Comment:
In other conventions, we have certain recommendation/guidelines for which spans should be created (e.g., [CloudEvents](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/trace/semantic_conventions/cloudevents.md#spans). So I wonder if we shouldn't have something similar here.

At least to me, from the text above it's not clear if instrumentation SHOULD always create a separate span for Elasticsearch client requests or if they should just add the attributes here to any existing HTTP client spans. Wouldn't the lack of such guidelines also create a inconsistency between language instrumentation like we have now?

@estolfo sorry if this was discussed somewhere already, but I couldn't find it.

Reply:
Hi @joaopgrassi thanks for your input. I purposely left it vague and didn't offer guidance on whether to create a separate http span because I think it depends on the implementation, what the user wants, and what backend/visualization the user is using.

What this spec is saying is that at least there will be one span with these attributes for an elasticsearch client request. There may or may not be a http span as well, but its existence doesn't change these semantic conventions.

I think the best approach, ultimately, would be to expose a config option to create separate http spans or just have the elasticsearch spans but I'm not sure that's something that should be mentioned in semantic conventions.

Reply:
Hi @joaopgrassi again, sorry for the back-and-forth but after talking internally, I think it actually might make sense to recommend that by default, no http span be created. Ultimately, what an Elasticsearch span represents is a db operation so it doesn't seem necessary to also have a network level span as well.
So what do you think about recommending a default of no http spans but suggesting a config for allowing the creation of them?

Also, we thought it might make sense to also add a `db.operation` attribute. This would correspond to the endpoint id, so it would be the "name" field in this spec: https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json

Reply:
Yes that's what I was trying to get at. With the current text I felt it was not clear if one should "mix" attributes with the HTTP span making the request to Elasticsearch or if a new one should always be created to model this "logical operation". 

@estolfo 

> I think it actually might make sense to recommend that by default, no http span be created

I'm not sure that's a good idea. From what you said and @Oberon00 then I think we simply need to make it more clear that instrumentations SHOULD create a span to model the Elasticsearch call and add the attributes as specified here. Then the question about creating/not creating underlying HTTP spans is up to the user if they add HTTP auto instrumentation or if the underlying HTTP client used by the Elasticsearch client library is also instrumented. What do you think?

Reply:
Yes that's what I was trying to get at. With the current text I felt it was not clear if one should "mix" attributes with the HTTP span making the request to Elasticsearch or if a new one should always be created to model this "logical operation". 

@estolfo 

> I think it actually might make sense to recommend that by default, no http span be created

I'm not sure that's a good idea. From what you said and @Oberon00 then I think we simply need to make it more clear that instrumentations SHOULD create a span to model the Elasticsearch call and add the attributes as specified here. No mention about any other span. 

Then the question about creating/not creating underlying HTTP spans is up to the user if they add HTTP auto instrumentation or if the underlying HTTP client used by the Elasticsearch client library is also instrumented. What do you think?

Reply:
ok, so if I understand the suggestion at this point: the spec should suggest that a span be created for an Elasticsearch request with the attributes described and not mention anything about an http span?

Reply:
Yeah I'd say that would makes the most sense, and if I also got it right, it's the direction @Oberon00  was going at with this previous comment above.

Reply:
Hi @joaopgrassi, I updated the description to remove a reference to http spans. 
I also added `db.operation` to the span attributes to capture the generic name of the request. It's the "name" field in the json schema [here](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json)

Reply:
I would still prefer some sort of "SHOULD" statement indicating that when talking to Elasticsearch such spans should be created, but I guess being vague is also fine 👍.

Reply:
I understand but it feels a bit strange to describe that a span should be created in the semantic conventions doc itself-- the existence of the semantic conventions doc implies that a span should be created.